alfalfa-import utility created.

sadjad opened the pull request.

shacklettbp commented on src/decoder/frame_info.cc:
```diff
@@ -0,0 +1,111 @@
+  const TargetHash & target_hash )
+{
+  stringstream ss;
+  ss << source_hash.str() << "#" << target_hash.str();
```
source_hash.str() + "#" + target_hash.str()

sadjad replied:
Done.

shacklettbp commented on src/decoder/frame_info.hh:
```diff
@@ -0,0 +1,65 @@
+  size_t offset;
+  size_t length;
+  SourceHash source_hash;
+  TargetHash target_hash;
```
This breaks the convention throughout the code of having all class member variables end with _. Can you just change these to getters and make them private?

sadjad replied:
Done.

shacklettbp commented on src/frontend/Makefile.am:
```diff
@@ -23,3 +23,6 @@ frameplayer_LDADD = ../decoder/libalfalfadecoder.a ../encoder/libalfalfaencoder.
 collisions_LDADD = ../decoder/libalfalfadecoder.a ../encoder/libalfalfaencoder.a ../display/libalfalfadisplay.a ../util/libalfalfautil.a $(X264_LIBS)
+
+alfalfa_import_SOURCES = alfalfa-import.cc
+alfalfa_import_LDADD = ../format/libalfalfavideo.a ../format/protobufs/libalfalfaprotobufs.a ../decoder/libalfalfadecoder.a ../encoder/libalfalfaencoder.a ../util/libalfalfautil.a $(X264_LIBS) $(PROTOBUF_LIBS) -lboost_system -lboost_filesystem
```
:(

sadjad replied:
Got rid of boost-filesystem ;)

sadjad commented:
There are some necessary changes that Deepak and I going to make, and I'm going to close this pull request. I will make a new pull request as soon as they're done (hopefully very soon!).

keithw commented:
Ok! Also: let's not commit auto-generated protobuf files into the repository. Instead, they should be built as part of the build system. (See https://github.com/ravinet/mahimahi/tree/master/src/protobufs for how to do this.)

Otherwise we have a problem where our code will stop compiling once the user upgrades their version of protobuf...

sadjad commented:
They're not, they're in the .gitignore! I think I used the exact same approach as Mahimahi. https://github.com/sadjad/alfalfa/blob/e035117d25e83c1409532985b16f2a9dcf27d67e/.gitignore

keithw commented:
Whoops, apparently I read too fast!

sadjad commented:
We changed the way we were storing databases to make them ready to create `alfalfa-combine` and other necessary commands. We also squashed all the commits together.
